libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/StoredCertificate.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.ssl;
+
+import org.elasticsearch.core.Nullable;
+
+import java.security.cert.X509Certificate;
+import java.util.Objects;
+
+/**
+ * Information about a certificate that is locally stored.It includes a reference to the {@link X509Certificate} itself,
+ * as well as information about where it was loaded from.
+ */
+public class StoredCertificate {
```

Comment:
This is used for the `_ssl/certificates` API.
That API uses a `CertificateInfo` which depends on `ToXContent` etc, which isn't available inside this library. So this class is a lightweight representation of the information needed for that API, and X-Pack will convert from this class to the X-Pack class.